python/ray/experimental/data/datasource/file_based_datasource.py

```diff
@@ -144,6 +144,10 @@ def _resolve_paths_and_filesystem(
 
     resolved_paths = []
     for path in paths:
+        if filesystem is not None:
+            # If we provide a filesystem, _resolve_filesystem_and_path will not
+            # slice off the protocol from the provided URI/path when resolved.
+            path = _unwrap_protocol(path)
```

Comment:
Is this different from filesystem.normalize_path()?

Reply:
Unfortunately yes. that doesn't strip off the protocol:
```python
fs.normalize_path("s3://foo/bar") == "s3://foo/bar"
```
Meanwhile, `FileSystem.from_uri("s3://foo/bar)` will return `"foo/bar"`, along with a newly constructed `S3FileSystem`, which seemed wasteful.